frontend/components/LogoutButton.jsx

```diff
@@ -0,0 +1,22 @@
+import { auth } from '../firebase/config';
+import { signOut } from 'firebase/auth';
+import { useRouter } from 'next/navigation';
+import Cookies from 'js-cookie';
+
+const Logout = () => {
+    const router = useRouter();
+
+    const handleLogout = async () => {
+        await signOut(auth);
+        Cookies.set('uid', 'false');
```

Comment:
having the value set to 'false', implies that when there is a uid it will be 'true'. Even though it is a string and not a boolean, it might feel misleading in the future. You can delete the cookie when the user logs outs, or you can set it to an empty string which is also a falsey value.